.changelog/17702.txt

```diff
@@ -0,0 +1,7 @@
+```release-note:bug
+resource/aws_glue_catalog_table: Add nil check for `schema_reference`
```

Comment:
To make this note more practitioner friendly, recommend saying what the change is doing for them:

```suggestion
resource/aws_glue_catalog_table: Prevent `schema_reference` attribute error
```